src/main/java/com/linkedin/kmf/apps/SingleClusterMonitor.java

```diff
@@ -71,11 +73,26 @@ public SingleClusterMonitor(Map<String, Object> props, String name) throws Excep
   @Override
   public void start() {
     _topicManagementService.start();
-    CompletableFuture<Void> completableFuture = _topicManagementService.topicManagementResult();
-    completableFuture.thenRun(() -> {
+    CompletableFuture<Void> topicPartitionResult = _topicManagementService.topicPartitionResult();
+    try {
+      /* Delay 2 second to reduce the chance that produce and consumer thread has race condition
+      with TopicManagementService and MultiClusterTopicManagementService */
+      long threadSleepMs = 1000 * 2;
+      Thread.sleep(threadSleepMs);
+    } catch (InterruptedException e) {
+      LOG.error("InterruptedException while thread sleeping.", e);
+    }
+    CompletableFuture<Void> topicPartitionFuture = topicPartitionResult.thenRun(() -> {
       _produceService.start();
       _consumeService.start();
     });
+
+    try {
+      topicPartitionFuture.get();
+    } catch (InterruptedException | ExecutionException e) {
+      LOG.error("Exception occurred while getting the TopicPartitionFuture", e);
```

Comment:
It's similar to my another comment on the `ConsumerFactoryImpl` constructor. If there is exception being thrown while waiting on the the producerService and consumer service to start, we should probably report/throw the exception to the caller to indicate that there is some problem occurred instead of just log the exception and then log `/SingleClusterMonitor started.` since it might not have started at all.

Reply:
We can throw Exception and add Exception signature to the App-level start() method.